openpype/pipeline/colorspace.py

```diff
@@ -221,8 +274,13 @@ def _get_context_settings(
 
 
 def get_imageio_file_rules_colorspace_from_filepath(
-    filepath, host_name, project_name,
-    config_data=None, file_rules=None,
+    filepath,
+    host_name,
+    project_name,
```

Comment:
We're allowing to 'fall back' onto the `get_current_context` for asset name and task name if not provided (when `None`).

`get_current_context()` also returns the project name. Why do we not allow `project_name` to also be `None` and then default to the context as well?

Similarly for `host_name` - why can't we rely on current context if not provided, similar to `get_colorspace_name_from_filepath` does.

Actually, I feel like each of these functions could have the same entry point for host name, project name, asset name and task name - allowing a fallback to current context in the same way.

We might also benefit by doing that in a dedicated function:

```
project_name, asset_name, task_name, host_name = _resolve_context(
    project_name,
    asset_name,
    task_name,
    host_name
)
```

Then the function would be:
```python
def _resolve_contexts(project_name, asset_name, task_name, host_name):
    """Resolve arguments that are None to current context"""
    if not all([project_name, asset_name, task_name, host_name]):
        context = get_current_context()
        project_name = project_name or context["project_name"]
        asset_name = asset_name or context["asset_name"]
        task_name = task_name or context["task_name"]
        host_name = host_name or context["host_name"]
    return project_name, asset_name, task_name, host_name
```

Just to ensure that each function would do the same. Even better we could _even_ use a function decorator for that? E.g. `@default_to_current_context` so that we do not even need to duplicate the resolving of those vars, etc. in each function.

Reply:
Already mentioned here https://github.com/ynput/OpenPype/pull/5825/files#r1433883103 .

For autofill with current context should be dedicated function.
With autofil you can't call the function with different partial context. Let's say I want to get colorspace for different asset but without task name, in that case the task name would be autofilled with completelly different value.
Also from the point of api functionality I don't know what the function actually requires to be functional. This makes sense if you want to call it from a host like Nuke, but it's not much intuitive if I would call out of any host. In that case I would have to know implementation of the function, to know what I have to pass in, because there is no current context.